lib/test.js

```diff
@@ -457,6 +462,20 @@ class Test {
 		this.timeoutTimer = null;
 	}
 
+	teardown(callback) {
```

Comment:
`addTeardown` maybe? We can be a bit more explicit here, as opposed to the `t` object which should be terser.

Reply:
All for. Wouldn't the same logic apply to for example `timeout()`? `setTimeout` or similar?

Reply:
Since it's an end user facing API, should we type check the input? I.e. ensure it's a function being passed in? As far as I can tell, something similar is not done to e.g. `timeout`, but could still be an improvement.

Reply:
We do that in some places, usually where there's ambiguity. No harm to doing that here too, I suppose.